detox/test/index.ios.js

```diff
@@ -10,46 +10,97 @@ import {
   StyleSheet,
   Text,
   View,
+  ScrollView,
   TouchableOpacity
 } from 'react-native';
+import _ from 'lodash';
+
+const STRESSFUL_STRING_LENGTH = 7800
+const STRESSFUL_EVENTS_COUNT = 150
+
+function buildStringByLength(length) {
+  str = ""
+  charcode = 65
+  for (let i=0; i < length; i++) {
+    str += String.fromCharCode(charcode)
+    charcode ++
+    if (charcode == 91) charcode = 65
+  }
+  return str
+}
 
 class example extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      greeting: undefined
+      greeting: undefined,
+      passToBridge: undefined
     };
   }
+
+  renderTestButton(label, onPress) {
+    return (
+      <TouchableOpacity onPress={onPress}>
+        <Text style={{color: 'blue', marginBottom: 20}}>{label}</Text>
+      </TouchableOpacity>
+    )
+  }
+
   render() {
     if (this.state.greeting) return this.renderAfterButton();
     return (
       <View style={{flex: 1, paddingTop: 20, justifyContent: 'center', alignItems: 'center'}}>
         <Text style={{fontSize: 25, marginBottom: 30}}>
           Welcome
         </Text>
-        <TouchableOpacity onPress={this.onButtonPress.bind(this, 'Hello')}>
-          <Text style={{color: 'blue', marginBottom: 20}}>Say Hello</Text>
-        </TouchableOpacity>
-        <TouchableOpacity onPress={this.onButtonPress.bind(this, 'World')}>
-          <Text style={{color: 'blue', marginBottom: 20}}>Say World</Text>
-        </TouchableOpacity>
+        {this.renderTestButton('Say Hello', this.onButtonPress.bind(this, 'Hello'))}
+        {this.renderTestButton('Say World', this.onButtonPress.bind(this, 'World'))}
+        {this.renderTestButton('Bridge Stress', this.bridgeStressButtonPressed.bind(this, 'Hello World'))}
+        {this.renderTestButton('Events Stress', this.eventsStressButtonPressed.bind(this, 'Hello World'))}
       </View>
     );
   }
   renderAfterButton() {
     return (
       <View style={{flex: 1, paddingTop: 20, justifyContent: 'center', alignItems: 'center'}}>
-        <Text style={{fontSize: 25}}>
-          {this.state.greeting}!!!
-        </Text>
+        <ScrollView>
+          <Text style={{fontSize: 10, width: 300, height: 10}}>
+            Bridge: {this.state.passToBridge}
+          </Text>
+          <Text style={{fontSize: 25}}>
+            {this.state.greeting}!!!
+          </Text>
+        </ScrollView>
       </View>
     );
   }
+
   onButtonPress(greeting) {
     this.setState({
       greeting: greeting
     });
   }
+
+  bridgeStressButtonPressed(greeting) {
+    this.onButtonPress(greeting)
```

Comment:
don't have 2 calls to `setState`, combine them to one call together, no need to call `onButtonPress`, just do the combined `setState` here. I want to make sure we send both things together over the bridge